src/aurora.ts

```diff
@@ -279,36 +279,40 @@ export class Aurora extends Construct {
         }),
       );
 
-      if (!props.skipUserProvisioning) {
-        const rdsUser = rdsUserProvisioner(user, {
-          userSecretArn: secret.secretArn,
-          dbName: props.defaultDatabaseName,
-          isWriter: userStr == 'writer', // good enough for now.
-        });
-        rdsUser.node.addDependency(this.cluster);
-      }
-
       if (!props.skipAddRotationMultiUser) {
         const rotation = this.cluster.addRotationMultiUser(user.pascal, { secret });
         // https://github.com/aws/aws-cdk/issues/18249#issuecomment-1005121223
         const sarMapping = rotation.node.findChild('SARMapping') as CfnMapping;
         sarMapping.setValue('aws', 'semanticVersion', passwordRotationVersion);
       }
-      return secret;
+      return { userStr, secret };
     });
+    this.secrets = secrets.map((s) => s.secret);
 
     if (!props.skipProxy) {
       this.proxy = new aws_rds.DatabaseProxy(this, 'Proxy', {
         dbProxyName: id.pascal,
         proxyTarget: aws_rds.ProxyTarget.fromCluster(this.cluster),
         //requireTLS: true, // If we're never allowing connections from outside the VPC, why bother?
-        secrets: this.secrets,
+        secrets: secrets.map((s) => s.secret),
```

Comment:
Do we need to do this again after line 290 above?

Reply:
Yeah do we not have access to the `this` object in this line or am I missing something?